browser/file_persister.go

```diff
@@ -0,0 +1,105 @@
+package browser
+
+import (
+	"fmt"
+	"net/url"
+	"strings"
+
+	"github.com/grafana/xk6-browser/env"
+	"github.com/grafana/xk6-browser/storage"
+)
+
+type presignedURLConfig struct {
+	getterURL string
+	headers   map[string]string
+	basePath  string
+}
+
+// newFilePersister will return either a persister that persists file to the local
+// disk or uploads the files to a remote location. This decision depends on whether
+// the K6_BROWSER_SCREENSHOTS_OUTPUT env var is setup with the correct configs.
+func newFilePersister(envLookup env.LookupFunc) (filePersister, error) {
+	envVar, ok := envLookup(env.ScreenshotsOutput)
+	if !ok || envVar == "" {
+		return &storage.LocalFilePersister{}, nil
+	}
+
+	popts, err := parseEnvVar(envVar)
+	if err != nil {
+		return nil, fmt.Errorf("parsing %s: %w", env.ScreenshotsOutput, err)
+	}
+
+	return storage.NewRemoteFilePersister(popts.getterURL, popts.headers, popts.basePath), nil
+}
+
+// parseEnvVar will parse a value such as:
+// url=https://127.0.0.1/,basePath=/screenshots,header.1=a,header.2=b
+// and return them.
+//
+//nolint:funlen,cyclop
+func parseEnvVar(envVarValue string) (presignedURLConfig, error) {
+	ss := strings.Split(envVarValue, ",")
+
+	var (
+		preSignedURLGetterURL string
+		basePath              string
+		headers               = make(map[string]string)
+	)
+	for _, s := range ss {
+		// The key value pair should be of the form key=value, so split
+		// on '=' to retrieve the key and value separately.
+		kv := strings.Split(s, "=")
+		if len(kv) <= 1 || len(kv) > 2 {
```

Comment:
?

```suggestion
		if len(kv) != 2 {
```

Reply:
🤦 yep. Changed in 9d293a06ec75da0170a104c04b19198662155da7